homeassistant/components/zwave_js/api.py

```diff
@@ -278,3 +285,30 @@ async def remove_from_device_registry(
         return
 
     registry.async_remove_device(device.id)
+
+
+class DumpView(HomeAssistantView):
+    """View to dump the state of the Z-Wave JS server."""
+
+    url = "/api/zwave_js/dump"
+    name = "api:zwave_js:dump"
+
+    @RequestDataValidator({"config_entry_id": str})
+    async def post(self, request: web.Request, data: dict) -> web.Response:
+        """Dump the state of Z-Wave."""
+        hass = request.app["hass"]
+
+        if data["config_entry_id"] not in hass.data[DOMAIN]:
+            raise http_exceptions.HttpBadRequest
+
+        entry = hass.config_entries.async_get_entry(data["config_entry_id"])
+
+        msgs = await dump.dump_msgs(entry.data[CONF_URL], async_get_clientsession(hass))
+
+        return web.Response(
+            body="\n".join(json.dumps(msg) for msg in msgs) + "\n",
```

Comment:
Should we use an f-string instead of string concatenation?

Reply:
No. That looks really bad.

Reply:
Then maybe break up the expression?

Reply:
Nevermind. Let's keep this.